src/transaction.ts

```diff
@@ -1822,9 +1823,33 @@ export class Transaction extends Dml {
 
     this._queuedMutations = [];
     this._options = {readWrite: options};
+    this._options.isolationLevel =
+      protos.google.spanner.v1.TransactionOptions.IsolationLevel.ISOLATION_LEVEL_UNSPECIFIED;
     this.requestOptions = requestOptions;
   }
 
+  setReadWriteTransactionOptions(options: RunTransactionOptions | undefined) {
+    /**
+     * Set optimistic concurrency control for the transaction.
+     */
+    if (options?.optimisticLock) {
+      this._options.readWrite!.readLockMode = ReadLockMode.OPTIMISTIC;
+    }
+    /**
+     * Set option excludeTxnFromChangeStreams=true to exclude read/write transactions
+     * from being tracked in change streams.
+     */
+    if (options?.excludeTxnFromChangeStreams) {
+      this._options.excludeTxnFromChangeStreams = true;
+    }
+    /**
+     * Set isolation level .
+     */
+    if (options?.isolationLevel) {
```

Comment:
how would this use the default isolation level if available ?

Reply:
I have updated the code
in the `Transaction` class constructor we are assigning the default isolationLevel value as `this._options.isolationLevel = IsolationLevel.ISOLATION_LEVEL_UNSPECIFIED;`

now here in this case `this._options.isolationLevel = options?.isolationLevel? options?.isolationLevel:this._getSpanner().defaultTransactionOptions.isolationLevel;`

if the `isolationLevel` will get pass at the transaction level it will assign that value to the `this._options.isolationLevel` otherwise this value `this._getSpanner().defaultTransactionOptions.isolationLevel ` will get assign which if being passed from spanner options will use that value otherwise it will also be containing default value of unspecified